stock_manual_transfer/models/stock_manual_transfer.py

```diff
@@ -0,0 +1,121 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockManualTransfer(models.Model):
+    _name = "stock.manual_transfer"
+    _description = "Stock Manual Transfer"
+    _inherit = ["mail.thread"]
+
+    name = fields.Char(
+        required=True,
+        copy=False,
+        readonly=True,
+        index=True,
+        default=lambda self: _("New"),
+    )
+    warehouse_id = fields.Many2one(
+        "stock.warehouse",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        default=lambda self: self.env.user._get_default_warehouse_id(),
+    )
+    date_planned = fields.Datetime(
+        "Planned Date",
+        default=fields.Datetime.now,
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    route_id = fields.Many2one(
+        "stock.location.route",
+        string="Preferred Route",
+        domain=[("manual_transfer_selectable", "=", True)],
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    transfer_line_ids = fields.One2many(
+        "stock.manual_transfer_line",
+        "transfer_id",
+        string="Transfer Lines",
+        copy=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+    )
+    state = fields.Selection(
+        string="Status",
+        selection=[
+            ("draft", "Draft"),
+            ("valid", "Validated"),
+        ],
+        default="draft",
+        required=True,
+        readonly=True,
+        copy=False,
+        tracking=True,
+    )
+    procurement_group_id = fields.Many2one("procurement.group", copy=False)
+    picking_ids = fields.One2many(
+        "stock.picking",
+        string="Transfers",
+        compute="_compute_picking_ids",
+    )
+
+    def action_validate(self):
+        route = self.env["stock.location.route"].browse(self.route_id.id)
+        if route.rule_ids.warehouse_id != self.warehouse_id:
```

Comment:
```suggestion
        if self.warehouse_id not in route.rule_ids.warehouse_id:
```

Because a route can have rules from different warehouses